pkg/kubernetes/kubernetes_test.go

```diff
@@ -0,0 +1,182 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package kubernetes
+
+import (
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/stretchr/testify/assert"
+
+	fake "k8s.io/client-go/kubernetes/fake"
+)
+
+const (
+	name        = "test"
+	namespace   = "testns"
+	label       = "testlabel"
+	servicename = "testservice"
+)
+
+var testclient = fake.NewSimpleClientset()
+var api = New(testclient)

```

Comment:
Instead of using global variable even in test code, it would be good to define newTestClient() method.

Reply:
Defined new method but still I have to share one test var so all tests can access to avoid creating new objects.